### PR TITLE
Add workflow to build wheels and publish to PyPI

### PR DESCRIPTION
**Pull request by larrybradley**
This PR is a first iteration to add a GitHub Actions workflow to automatically build wheels and publish to PyPI.  Obviously I can't test this locally, so it's likely to require some iteration.  

Also, I do not have admin permission to the repo -- can someone please add me as an admin?  I need that to add a PyPI token to this repo to enable automatic publishing.

CC: @pllim, @Russell-Ryan

### CODE REVIEW

**Comment by pllim** [comment]
Trying close/reopen trick

**Comment by larrybradley** [comment]
I think the wheel building workflow isn't running because it's from a PR on my fork.  I may need to merge this and manually trigger that job.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        - cp311*macosx_x86_64
+
+        - cp39*macosx_arm64
+        - cp310*macosx_arm64
+        - cp311*macosx_arm64
+
+        # Windows wheels
+        - cp39*win_amd64
+        - cp310*win_amd64
+        - cp311*win_amd64
```
Comment (pllim) [inline .github/workflows/publish.yml:61]: You are so brave to try building for all the OS...

Comment (larrybradley) [reply .github/workflows/publish.yml:61]: YOLO!  :smile: I'll remove them if they fail.

**Review by pllim — APPROVED** [review]
YOLO

**Comment by larrybradley** [comment]
FYI, the workflow is running now: https://github.com/spacetelescope/pypolyclip/actions/runs/5741867295

🤞

**Comment by larrybradley** [comment]
All of the wheels passed (linux, win64, macos intel and arm)!

@duytnguyendtn Can you please add me as an admin to the repo so that I may add the PyPI token?

**Comment by duytnguyendtn** [comment]
Should be done, try now @larrybradley

**Comment by larrybradley** [comment]
Thanks, @duytnguyendtn!

**Comment by larrybradley** [comment]
FYI, I'm going to do an initial alpha release to reserve the name on PyPI and test the publish workflow.

**Pull request merged**